### PR TITLE
remove cereal, make auth a real module

### LINKED ISSUE #139: Remove dependency on cereal

### PR DESCRIPTION
**Pull request by martyall**
fixes #139

### CODE REVIEW

File: hs-abci-sdk/src/Tendermint/SDK/Crypto.hs [hunk 1]
```diff
@@ -79,7 +79,14 @@ instance RecoverableSignatureSchema Secp256k1 where
     recover _ sig dig = Secp256k1.recover sig (msgFromSHA256 dig)
     -- NOTE: I think the use of Data.Serialize is harmless here, because it basically
```
Comment (IvantheTricourne) [inline hs-abci-sdk/src/Tendermint/SDK/Crypto.hs]: Probably don't need this comment anymore then?

File: hs-abci-sdk/package.yaml [hunk 2]
```diff
@@ -150,12 +150,12 @@ tests:
     - -with-rtsopts=-N
 
     dependencies:
+      - cereal
```
Comment (IvantheTricourne) [inline hs-abci-sdk/package.yaml:153]: So cereal is only used for some tests?

**Review by IvantheTricourne — COMMENTED** [review]
Seems legit. Does simple storage really need cereal still?

**Comment by martyall** [comment]
I think it's fine for tests to use `cereal`. Simple storage does because it's really convenient, and is effectively a test, idk. In order to remove it I need to put it's storage types in a `.proto` file, which in this case was just an `Int32`, it seemed lame.

**Review by charlescrain — APPROVED** [review]

**Pull request merged**